app/scripts/modules/core/src/search/global/GlobalSearch.tsx

```diff
@@ -0,0 +1,409 @@
+              <span className="error-message">Please enter at least 3 characters</span>
+            </li>
+          </ul>
+        }
+        {!!query && querying &&
+          <ul className="dropdown-menu" role="menu">
+            <li className="horizontal middle center spinner-section">
+              <Spinner size="small"/>
+            </li>
+          </ul>
+        }
+        {!query && showRecentItems &&
+          <RecentlyViewedItems
+            limit={5}
+            Component={({ results: recentCategories }) => {
+              if (!recentCategories.length) {
+                return null;
+              }
+
+              this.resultRefs = range(recentCategories.length).map(() => []);
+
+              return (
+                <ul className="dropdown-menu" role="menu">
+                  {recentCategories.map((category, categoryIndex) => ([
+                    <li key={category.category} className="category-heading">
+                      <div className="category-heading">Recent {category.category}</div>
+                    </li>,
+                    ...category.results.map((result, index) => {
+                      const params = result.params || {};
+                      const account = result.account || params.account || params.accountId || params.accountName;
+
+                      return (
+                        <li
+                          key={result.id}
+                          className="result"
+                          onClick={() => {
+                            this.hideResults();
+                            ReactGA.event({ category: 'Global Search', action: 'Recent Item Selected' });
+                          }}
+                        >
+                          <UISref to={result.state} params={result.params}>
+                            <a
+                              ref={(ref) => {
+                                if (this.resultRefs[categoryIndex]) {
+                                  this.resultRefs[categoryIndex][index] = ref;
+                                }
+                              }}
+                              onKeyDown={this.navigateResult}
+                            >
+                              <SearchResult displayName={result.displayName} account={account} />
+                            </a>
+                          </UISref>
+                        </li>
+                      );
+                    })
+                  ]))}
+                </ul>
+              );
+            }}
+          />
+        }
+        {!!query && !querying && showSearchResults &&
+          <ul className="dropdown-menu" role="menu">
+            {categories.map((category, categoryIndex) => ([
+              <li key={category.type.id} className="category-heading">
+                <div className="category-heading">{category.type.displayName}</div>
+              </li>,
+              category.results.map((result, index) => (
+                <li className="result">
```

Comment:
Need a `key`

Reply:
thx good catch 💯

Reply:
also, just figured out why i didn't get a console warning — this is a sub-array rather than a top-level children array.

Reply:
relevant: https://github.com/facebook/react/issues/9168#issuecomment-334247132